src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { State } from './types';
+
+const defaultState: State = {
+  shipmentId: 0,
+  toAddress: '',
+  fromAddress: '',
+  toZipCode: '',
+  fromZipCode: '',
+  weight: 0,
+  marks: [],
+};
+
+class Gui {
+  state: State;
+  constructor() {
+    this.state = defaultState;
+  }
+
+  // on(eventType: string): void;
+
+  // trigger(eventType: string, state: Shipment): any {}
+
+  getState(): State {
+    return this.state;
+  }
+}
+
+class Client {
+  gui;
+
+  constructor(gui: Gui) {
+    this.gui = gui;
+  }
+
+  onShip(shipment: Shipment) {}
```

Comment:
Why there's no logic in this method? It should call the ship() method and log the result in console.

Reply:
done